cmake/modules/SearchInstalledSoftware.cmake

```diff
@@ -37,6 +37,21 @@ if(NOT builtin_zlib)
 endif()
 if(builtin_zlib)
   message(STATUS "Building zlib included in ROOT itself")
+  if((CMAKE_SYSTEM_PROCESSOR MATCHES "amd64|x86_64|AMD64|x86|X86") AND (CMAKE_SYSTEM_NAME MATCHES "Linux"))
```

Comment:
This may break on 32bit systems. Is SIMD code in zlib really portable between 32 and 64 bits?

Reply:
@amadio it should not break 32b systems, because my idea was to enable SIMD only for 64b..

Reply:
The check above will be true on 32 bit systems (x86).

Reply:
@amadio hi, I checked that it is actually equal to output of  ``` uname -p ``` (https://cmake.org/cmake/help/v3.7/variable/CMAKE_HOST_SYSTEM_PROCESSOR.html#variable:CMAKE_HOST_SYSTEM_PROCESSOR)

```
[sftnight@lcgapp-slc6-i386-1 ~]$ uname -p
i686
```

Reply:
My machine says:
```
$ uname -p
Intel(R) Core(TM) i7-6700 CPU @ 3.40GHz
```
This means that despite my machine being Intel 64 bit, the SIMD support will not be enabled at all.

In any case, `i686` and `x86` are for 32 bit systems. On a 64bit system, it should say `x86_64` instead. 
```
$ arch
x86_64
```
The command above will also say `i686` in your case, most likely (same as `uname -m`).

Reply:
@amadio :) true!
```
[sftnight@lcgapp-slc6-i386-1 ~]$ uname -p
i686
[sftnight@lcgapp-slc6-i386-1 ~]$ arch
i686
[sftnight@lcgapp-slc6-i386-1 ~]$ uname -m
i686
```
but for safety i will fix it